## 10

In [6]:
count = 0
with open('popular-names.txt') as f:
    for line in f:
        count += 1
print(count)

2740


>＄ wc -l popular-names.txt <br>
>2740 popular-names.txt

## 11

In [17]:
def print_file(fp,n):
    with open(fp,'r') as f:
        for i in range(n):
            print(f.readline(),end="" )

with open('popular-names.txt','r') as f,open('change-popular-names.txt','w',newline="\n") as ch:
    for line in f:
        ch.write(line.replace('\t',' '))
print_file('change-popular-names.txt',10)

Mary F 7065 1880
Anna F 2604 1880
Emma F 2003 1880
Elizabeth F 1939 1880
Minnie F 1746 1880
Margaret F 1578 1880
Ida F 1472 1880
Alice F 1414 1880
Bertha F 1320 1880
Sarah F 1288 1880


>＄ cat popular-names.txt | tr "\t" " ">test.txt <br>
>＄ diff -s change-popular-names.txt test.txt <br>
>Files change-popular-names.txt and test.txt are identical

## 12

In [19]:
with open('popular-names.txt','r') as f,open('col1-popular-names.txt','w',newline="\n") as ch1,open('col2-popular-names.txt','w',newline="\n") as ch2:
    for line in f:
        ch1.write(line.split('\t')[0]+'\n')
        ch2.write(line.split('\t')[1]+'\n')

>＄ cut -f 1 -d " " change-popular-names.txt > test.txt<br>
>＄ diff -s test.txt col1-popular-names.txt<br>
>Files test.txt and col1-popular-names.txt are identical<br>
>＄ cut -f 2 -d " " change-popular-names.txt > test.txt<br>
>＄ diff -s test.txt col2-popular-names.txt<br>
>Files test.txt and col2-popular-names.txt are identical<br>

## 13

In [21]:
count=0
with open('col1-popular-names.txt','r') as ch1:
    for line in ch1:
        count+=1
with open('col1-popular-names.txt','r') as ch1,\
     open('col2-popular-names.txt','r') as ch2,\
     open('col3-popular-names.txt','w',newline="\n") as ch3:
    for line in range(count):
        ch3.write(ch1.readline().replace('\n','\t')+ ch2.readline())

>＄ paste col1-popular-names.txt col2-popular-names.txt > test.txt<br>
>＄ diff -s test.txt col3-popular-names.txt<br>
>Files test.txt and col3-popular-names.txt are identical<br>

## 14

In [4]:
def print_line(n):
    N=int(n)
    with open('popular-names.txt') as f:
        for i in range(N):
            print(f.readline(),end='')

#import sys
#args = sys.argv
print_line(2)

Mary	F	7065	1880
Anna	F	2604	1880


>＄ head -n 2 popular-names.txt<br>
>Mary    F       7065    1880<br>
>Anna    F       2604    1880<br>

## 15

In [2]:
def print_last_line(n):
    N=int(n)
    count=0
    with open('popular-names.txt') as f:
        for line in f:
            count+=1
    with open('popular-names.txt') as f:
        for i in range(count):
            if i<count-N:
                f.readline()
            else:
                print(f.readline(),end='')


#import sys
#args = sys.argv
print_last_line(2)

Elijah	M	13764	2016
Ethan	M	13758	2016


>＄ tail -n 2 popular-names.txt<br>
>Elijah  M       13764   2016<br>
>Ethan   M       13758   2016<br>

## 16

In [24]:
def dev_setntence(n):
    N=int(n)
    count=0
    with open('popular-names.txt') as f:
        for line in f:
            count+=1
    with open('popular-names.txt') as f:
        for i in range(N-1):
            with open('dev'+str(i+1)+'-popular-names.txt','w',newline="\n") as d:
                for line in range(count//N):
                    d.write(f.readline())
        with open('dev'+str(N)+'-popular-names.txt','w',newline="\n") as d:
            for line in f:
                d.write(line)


# import sys
# args = sys.argv
dev_setntence(3)

>＄ split -n l/3 popular-names.txt

## 17

In [31]:
names = []
with open('popular-names.txt') as f:
    for line in f:
        names.append(line.split('\t')[0])
names=sorted(set(names))
with open('names.txt','w',newline="\n") as f:
    for name in names:
        f.write(name+"\n")

>＄ cut -f 1 -d " " change-popular-names.txt | sort |uniq >test.txt<br>
>＄ diff -s test.txt names.txt<br>
>Files test.txt and names.txt are identical

## 18

In [11]:
li = []
with open('popular-names.txt') as f,\
     open('sort3-popular-names.txt','w',newline="\n") as s:
    for line in f:
        li.append(line)
    li=sorted(li,key=lambda x: int(x.split('\t')[2]),reverse=True)
    for line in li:
        s.write(line)

>＄ sort -n -r -k 3 popular-names.txt >test.txt

## 19

In [42]:
dic = {}
with open('popular-names.txt') as f:
    for line in f:
        dic.setdefault(line.split('\t')[0],0)
        dic[line.split('\t')[0]]+=1
line=dic.items()
line=sorted(line,key=lambda x: x[1],reverse=True)
for ans in line[:10]:
    print(ans)

('James', 116)
('William', 109)
('John', 108)
('Robert', 108)
('Mary', 92)
('Charles', 75)
('Michael', 74)
('Elizabeth', 73)
('Joseph', 71)
('Margaret', 60)


>＄ cut -f 1 -d " " change-popular-names.txt | sort | uniq -c| sort -n -r -k 1 | head<br>
>    116 James<br>
>    109 William<br>
>    108 Robert<br>
>    108 John<br>
>     92 Mary<br>
>     75 Charles<br>
>     74 Michael<br>
>     73 Elizabeth<br>
>     71 Joseph<br>
>     60 Margaret